tools/ci_connector_ops/ci_connector_ops/pipelines/utils.py

```diff
@@ -73,57 +110,78 @@ async def get_file_contents(container: Container, path: str) -> Optional[str]:
         return await container.file(path).contents()
     except QueryError as e:
         if "no such file or directory" not in str(e):
-            # this is the hicky bit of the stopgap because
             # this error could come from a network issue
             raise
     return None
 
 
-# This is a stop-gap solution to capture non 0 exit code on Containers
-# The original issue is tracked here https://github.com/dagger/dagger/issues/3192
+async def get_container_output(container: Container) -> Tuple[str, str]:
+    """Retrieve both stdout and stderr of a container, concurrently.
+
+    Args:
+        container (Container): The container to execute.
+
+    Returns:
+        Tuple[str, str]: The stdout and stderr of the container, respectively.
+
+    Raises:
+        ExecError: If the container exit code is not 0.
+    """
+    async with asyncer.create_task_group() as task_group:
+        soon_stdout = task_group.soonify(container.stdout)
+        soon_stderr = task_group.soonify(container.stderr)
+    return soon_stdout.value, soon_stderr.value
```

Comment:
This convenience is for when you need both `stdout` and `stderr`. This may not be necessary in a future release if this gets solved:
- https://github.com/dagger/dagger/issues/3496

It's preferable to use `container.stdout()` directly if you don't need the other and vice versa.

Note that this doesn't handle the `ExecError` because I prefer handling it where it's needed. That's why there's `get_exec_result`, it not only returns `exit_code` but also handles `ExecError` if that's desirable.